transport-native-epoll/src/main/java/io/netty/channel/epoll/NativeDatagramPacketArray.java

```diff
@@ -32,14 +33,17 @@
 /**
  * Support <a href="http://linux.die.net/man/2/sendmmsg">sendmmsg(...)</a> on linux with GLIBC 2.14+
  */
-final class NativeDatagramPacketArray implements ChannelOutboundBuffer.MessageProcessor {
+final class NativeDatagramPacketArray {
 
     // Use UIO_MAX_IOV as this is the maximum number we can write with one sendmmsg(...) call.
     private final NativeDatagramPacket[] packets = new NativeDatagramPacket[UIO_MAX_IOV];
 
     // We share one IovArray for all NativeDatagramPackets to reduce memory overhead. This will allow us to write
     // up to IOV_MAX iovec across all messages in one sendmmsg(...) call.
     private final IovArray iovArray = new IovArray();
+
+    private final MyMessageProcessor processor = new MyMessageProcessor();
```

Comment:
why introducing this? The original solution was avoiding this pointer chasing

Reply:
@franz1981 because I did not want to create a new object all the time... or you say I should just still implement it directly and update the fields accordantly ?

Reply:
> or you say I should just still implement it directly and update the fields accordantly 

Excatly :+1:

Reply:
I can do this but I think what we have is more elegant and clean... do you really think it matters ?

Reply:
Good question...I haven't measured so I cannot tell if is that relevant. but you risk to have this -> processor -> remote pointer chasing that would be saved by not allowing a mutable state (ie remote), although encapsulated in a different class.
`NativeDatagramPacketArray` is final and package private so I won't care that much to be that clean :P

Reply:
I think I would keep it like it is for now.. it just feels "cleaner" .